src/plugins/tables/tables.js

```diff
@@ -136,7 +136,7 @@ $document.on( "draw.dt", selector, function( event, settings ) {
 		li = document.createElement( "LI" );
 
 	// Determine if Pagination required
-	if ( paginate_buttons.length === 1 || ( pagination.find( ".previous, .next" ).length === 2 && paginate_buttons.length < 4 ) ) {
+	if ( paginate_buttons.length === 1 || ( ( pagination.find( ".last, .first" ).length === 2 || pagination.find( ".previous, .next" ).length === 2 ) && paginate_buttons.length === 3 ) || ( pagination.find( ".last, .first" ).length === 2 && paginate_buttons.length === 5 ) ) {
```

Comment:
Is it possible to simplify that conditional?

May be doing some variable caching to avoid to reevaluate the following expression

* ``` paginate_buttons.length ```
* ``` pagination.find( ".last, .first" ).length === 2```

I am suggesting to start the evaluation with ``` paginate_buttons.length ``` first then the other conditional. something like:
```
	paginate_buttons.length === 1
		|| 
	( 
		paginate_buttons.length === 3  &&
		( 
			pagination.find( ".last, .first" ).length === 2 || 
			pagination.find( ".previous, .next" ).length === 2 
		)
		
	)
		|| 

	( 
		paginate_buttons.length === 5 &&
		pagination.find( ".last, .first" ).length === 2 
	)
```

Reply:
Fixed